tests/test_downloader.py

```diff
@@ -508,10 +517,10 @@ def parse(self, response: scrapy.http.Response):
 
         crawler = make_crawler(TestSpider, {})
 
-        with warnings.catch_warnings(record=True) as warning_msg:
+        with pytest.warns(UserWarning) as record:
```

Comment:
Are these changes related to the pull request? 
The test docstring contradicts the test behavior.

Reply:
Yes, wherein it should not encounter particular warning messages by setting `not_existing=True`.

Reply:
aha, I see, thanks!